pandas/core/groupby/generic.py

```diff
@@ -1340,21 +1341,21 @@ def transform(self, func, *args, **kwargs):
         # optimized transforms
         func = self._get_cython_func(func) or func
 
-        if isinstance(func, str):
-            if not (func in base.transform_kernel_whitelist):
-                msg = "'{func}' is not a valid function name for transform(name)"
-                raise ValueError(msg.format(func=func))
-            if func in base.cythonized_kernels:
-                # cythonized transformation or canned "reduction+broadcast"
-                return getattr(self, func)(*args, **kwargs)
-            else:
-                # If func is a reduction, we need to broadcast the
-                # result to the whole group. Compute func result
-                # and deal with possible broadcasting below.
-                result = getattr(self, func)(*args, **kwargs)
-        else:
+        if not isinstance(func, str):
             return self._transform_general(func, *args, **kwargs)
 
+        elif func not in base.transform_kernel_whitelist:
```

Comment:
would it be possible to consoliate this logic into the base class to keep this DRY?

Reply:
im optimistic about this

Reply:
do you have a strong preference for doing this in this PR?  im trying to keep the scope narrow to avoid conflicting with #29124.

Reply:
no can do later